numba_dppy/dpnp_glue/dpnp_randomimpl.py

```diff
@@ -0,0 +1,1287 @@
+import numba_dppy.dpnp_glue.dpnpimpl as dpnp_ext
+from numba import types
+from numba.core.typing import signature
+from . import stubs
+import numba_dppy.dpnp_glue as dpnp_lowering
+from numba.core.extending import overload, register_jitable
+import numpy as np
+from numba_dppy import dpctl_functions
+import os
+
+
+@register_jitable
+def check_range(low, high):
+    if low >= high:
+        raise ValueError("Low cannot be >= High")
+
+
+@register_jitable
+def common_impl(low, high, res, dpnp_func, PRINT_DEBUG):
+    check_range(low, high)
+
+    sycl_queue = dpctl_functions.get_current_queue()
+    res_usm = dpctl_functions.malloc_shared(res.size * res.itemsize, sycl_queue)
+
+    dpnp_func(res_usm, low, high, res.size)
+
+    dpctl_functions.queue_memcpy(
+        sycl_queue, res.ctypes, res_usm, res.size * res.itemsize
+    )
+    dpctl_functions.free_with_queue(res_usm, sycl_queue)
+
+    dpnp_ext._dummy_liveness_func([res.size])
+
+    if PRINT_DEBUG:
+        print("DPNP implementation")
```

Comment:
It seems like code duplication.
Only calling for `dpnp_funct()` with different args are different.
Can we somehow eliminate duplication?

Reply:
Each of these are a different python object. Each object generates a different function call. I do not think these can be eliminiated.

Reply:
Moved to https://github.com/IntelPython/numba-dppy/issues/213